tests/installation/first_boot.pm

```diff
@@ -10,6 +10,12 @@ sub run() {
         return;
     }
 
+    # Wait for desktop to load after booting from an installed disk image
+    if ( check_var( "BOOT_HDD_IMAGE", 1 ) ) {
+        assert_screen "generic-desktop", 120;
+        return;
```

Comment:
I can't get the point why need to change first_boot.pm, isn't finish_desktop and console_setup should works like what sysauth test looking for(ie. detect desktop appears and switch to console?